### ISSUE TITLE
Plot2D no longer sorts coordinates before plotting

### ISSUE THREAD

**Issue by fmaussion** [pydata/xarray#2422]
I have a dataset with decreasing latitude coordinates.

With ``xarray`` v0.10.8, this is what happens when plotting:

![i1](https://user-images.githubusercontent.com/10050469/45765619-b58ceb00-bc35-11e8-94b4-5e37cb9f2a0a.png)

But on latest master the image is now upside down:

![i2](https://user-images.githubusercontent.com/10050469/45765681-e53bf300-bc35-11e8-95cc-8e4fb520bfe5.png)

Sorry if I missed a change along the way, I was off for a long time.

**Comment by shoyer** [comment]
The most likely source here is https://github.com/pydata/xarray/pull/2294

**Comment by fmaussion** [comment]
Looks like a difficult thing to test, also...

**Comment by dcherian** [comment]
Yes #2294 changed the default `xincrease, yincrease` to be `None` rather than `True`. Previously, if `xincrease` was `True` we were setting axis limits and this prevented the axis from auto-updating its limits. Now that we are using the `invert_axis()` method, this might not be an issue. (Actually, now that I look at the code for `invert_axis()`, all I needed to do earlier was set the kwarg `auto=None` in the call to `set_xlim`, so this would be an easy way to revert to the previous code.)

I switched the default to `None` because I thought it was better to make the least number of modifications to the axis by default.

That said, I cannot reproduce this with a dataset with decreasing latitude so I'm very confused about what's happening.
```
da = xr.DataArray(np.arange(24).reshape(3,8), dims=('latitude', 'longitude'),
                 coords={'latitude': [8, 0, -8],
                        'longitude': np.arange(8)})

da.plot()
```

PS: I'm on a ship in the middle of the Pacific with crappy internet access. @fmaussion I'd appreciate it if you could look at this so we can get the release out. I will respond but there'll be a large time lag.

**Comment by fmaussion** [comment]
> I cannot reproduce this with a dataset with decreasing latitude 

Maybe a MPL version issue? Because our docs are also afffected, see:
- http://xarray.pydata.org/en/stable/plotting.html#two-dimensions
- http://xarray.pydata.org/en/latest/plotting.html#two-dimensions

> @fmaussion I'd appreciate it if you could look at this

I Will have a look now

**Comment by shoyer** [comment]
I think this is fixed by #2425?
